ACP2E-2872: [Documentation] Update documentation regarding Bundle pro… (#307)
AdobeDocs/commerce-webapi: src/pages/rest/modules/bundle/index.md

---

title: Bundle module
description: Create and maintain bundle product type in the Magento application.
keywords:
- REST
---

# Bundle module

The `Bundle` module provides functionality for creating and maintaining bundle product types.

> ## Retrieve store specific bundle product details

Reviewer: ```suggestion
## Retrieve store-specific bundle product details
```

> We can retrieve store specific bundle product details for multi-store configuration
> 
> The following example uses the `GET V1/products/{sku}` call to get bundle product details. For store specific
> bundle product details add `store_code` at API request. Like this `GET {store_code}/V1/products/{sku}`

Reviewer: ```suggestion
The following example retrieves store-specific bundle product details for a multi-store configuration by using the `GET V1/products/{sku}` call. For store specific
bundle product details add the `store_code` to the API request. For example, `GET {store_code}/V1/products/{sku}`
```

        {
            "attribute_code": "shipment_type",
            "value": "0"
        },
        {
            "attribute_code": "options_container",
            "value": "container2"
        },
        {
            "attribute_code": "msrp_display_actual_price_type",
            "value": "0"
        },
        {
            "attribute_code": "url_key",
            "value": "desktop-computer"
        },
        {
            "attribute_code": "price_view",
            "value": "0"
        },
        {
            "attribute_code": "required_options",
            "value": "0"
        },
        {
            "attribute_code": "has_options",
            "value": "0"
        },
        {
            "attribute_code": "sku_type",
            "value": "0"
        },
        {
            "attribute_code": "price_type",
            "value": "0"
        },
        {
            "attribute_code": "tax_class_id",
            "value": "2"
        },
        {
            "attribute_code": "weight_type",
            "value": "0"
        },
        {
            "attribute_code": "category_ids",
            "value": [
                "32"
            ]
        }
    ]
}
```

> We can get `option_id` from bundle product options from  the response and use the `option_id` to update bundle product option.

Reviewer: ```suggestion

Retrieve the `option_id` from the bundle product options section of the response and use it to update the bundle product option.
```

Can you clarify what is being updated here and why.

> ## Update store specific bundle product options
> 
> The following example uses the `PUT /V1/bundle-products/options/{optionId}` call to update bundle product option. For store specific
> bundle product option update use `store_code` at API request. Like this `PUT {store_code}/V1/bundle-products/options/{optionId}`
> 

Reviewer: ```suggestion
## Update store-specific bundle product options

The following example uses the `PUT /V1/bundle-products/options/{optionId}` call to update the bundle product option. For a store-specific request, add the `store_code`. For example, `PUT {store_code}/V1/bundle-products/options/{optionId}`

```

**Sample Usage:**

`PUT <host>/rest/<store_code>/V1/bundle-products/options/{optionId}`

## Example

> The following example update bundle product option:

Reviewer: ```suggestion
The following example updates the bundle product option:
```

`PUT <host>/rest/default/V1/bundle-products/options/14`

<CodeBlock slots="heading, code" repeat="2" languages="JSON, JSON" />

#### Payload

```json
{
  "option_id": 14,
  "title": "RAM-Version-2",
  "required": true,
  "type": "select",
  "position": 1,
  "sku": "Desktop-Computer",
  "product_links": [
    {
      "id": "19",
      "sku": "RAM-12GB",
      "option_id": 14,
      "qty": 10,
      "position": 1,
      "is_default": false,
      "price": 25,
      "price_type": null,
      "can_change_quantity": 0,
      "extension_attributes": {}
    },
    {
      "id": "20",
      "sku": "RAM-24GB",
      "option_id": 14,
      "qty": 12,
      "position": 2,
      "is_default": false,
      "price": 50,
      "price_type": null,
      "can_change_quantity": 0,
      "extension_attributes": {}
    }
  ],
  "extension_attributes": {}
}
```

#### Response

```json
14
```

> ## Update store specific bundle product option links
> 
> The following example uses the `PUT /V1/bundle-products/{sku}/links/{optionId}` call to update bundle product option link. For store specific
> bundle product option link update use `store_code` at API request. Like this `PUT {store_code}/V1/bundle-products/{sku}/links/{optionId}`
> 

Reviewer: ```suggestion
## Update store-specific bundle product option links

The following example uses the `PUT /V1/bundle-products/{sku}/links/{optionId}` call to update the bundle product option link. For a store-specific call, add `store_code` to the API request. For example, `PUT {store_code}/V1/bundle-products/{sku}/links/{optionId}`

```

**Sample Usage:**

`PUT <host>/rest/<store_code>/V1/bundle-products/{sku}/links/{optionId}`

## Example

> The following example update bundle product option link:

Reviewer: ```suggestion
The following example updates the bundle product option link:
```

`PUT <host>/rest/default/V1/bundle-products/Desktop-Computer/links/14`

<CodeBlock slots="heading, code" repeat="2" languages="JSON, JSON" />

#### Payload

```json
{
  "linkedProduct":
  {
    "id": "19",
    "option_id": 14,
    "sku": "RAM-12GB",
    "qty": 2,
    "is_default": false,
    "can_change_quantity": 0,
    "price": 35,
    "price_type": 0,
    "extension_attributes": { }
  }
}
```

#### Response

```json
true

> ```

Reviewer: ```suggestion
```

```
